srml/eth-relay/src/tests.rs

```diff
@@ -1,16 +1,35 @@
 //! Tests for the module.
+use super::*;
+//use sr_primitives::traits::SignedExtension;
+//use support::{
+//	assert_err, assert_noop, assert_ok,
+//	traits::{Currency, ExistenceRequirement::AllowDeath, LockableCurrency, ReservableCurrency},
+//};
+//use system::RawOrigin;
 
-use std::str::FromStr;
+use mock::{EthRelay, ExtBuilder, System};
+//use mock::{info_from_weight, EthRelay, ExtBuilder, Runtime, System, CALL};
 
-use hex_literal::hex;
-use rustc_hex::FromHex;
+//use rstd::prelude::*;
 use sr_eth_primitives::{
 	receipt::{LogEntry, TransactionOutcome},
 	Address, Bloom, H64, U128,
 };
 
-use super::*;
-use mock::{EthRelay, ExtBuilder, System};
+use hex_literal::hex;
+use rustc_hex::FromHex;
+use sha3::Keccak256;
+use std::str::FromStr;
+use std::u64;
+use primitives::crypto::Ss58Codec;
+use node_primitives::AccountId;
+
+pub fn to_hex_string(bytes: Vec<u8>) -> String {
```

Comment:
Why not directly convert using:

```
let raw_sub_key : Vec<u8> = result.params[3].value.clone().to_bytes().expect("Param Convert to Bytes Failed.");

			let darwinia_account: AccountId32 = AccountId32::try_from(raw_sub_key.as_slice()).expect("Account Parse Failed.");
```

Reply:
Please refer: https://github.com/darwinia-network/darwinia/blob/develop/srml/eth-backing/src/lib.rs#L313

This test can also been kept as alternative sample for experimental purpose. cause ```String``` seems can not be used in substrate module.

Reply:
In fact `String` is `Vec<u8>`.

And yes, substrate use `Vec<u8>` to represent the `String`.

But I found `to_owned` can convert the `&str` to `String` in substrate module. https://github.com/darwinia-network/darwinia/blob/develop/srml/eth-backing/src/lib.rs#L239

Furthermore, we can run the test in `std`, `String` can be brought into this mod.

```rust
#[cfg(all(feature = "std", test))]
mod tests;
```